accepted/00000-maintenance-windows.md

```diff
@@ -0,0 +1,191 @@
+A "Create" button will create the new schedule. The ICalendar data are stored in the database.
+
+
+### Updating a Maintenance Schedule
+
+A Maintenance Schedule will change over the time. A Maintenance Window will be moved or removed on short notice,
+or a new Maintenance Window slot will be created.
+
+This happens while updating the ICalendar data of a maintenance schedule.
+
+While adding a new slot is a non-issue, the movement or removal of a window needs to be handled properly.
+
+As it is nearly impossible to identify a change of a special event we need to follow a different strategy.
+When we refresh or upload a changed ICalendar file, we need to re-evaluate all scheduled actions of systems using the
+new schedule.
+We also need to provide a resolve strategy. This can happen while implementing modules to move actions
+to a different time. Possible modules could be:
+
+- Nearest: Use the nearest maintenance window
+- Previous: Use the maintenance window before the scheduled date
+- Next: Use the maintenance window after the scheduled date
+- Cancel: Cancel the Action
+- Fail: do not import the new schedule at all and keep the existing one
+
+The customer can configure the order:
+
+  resolveOrder=Nearest,Previous,Next,Fail
+
+
+### Schedule an Action
+
+When a system is assigned to a schedule, only schedules withing a maintenance window should be allowed.
+In the User Interface we should show in these cases not the datepicker, but two simple Combo boxes with Date and Time.
+The Date contains the next 20 dates where a maintenance window exists. Depending on the date box the available
+times in the time box change.
+
+When using SSM it is possible that systems have different maintenance windows. The combination systems belong
+to one maintenance schedule and systems without maintenance schedule is not critical. The action would be scheduled
+at a time which is inside of a maintenance window of the systems which belong to the schedule.
+
+Systems in different schedules needs special handling:
+
+1. do not allow systems with different schedules in SSM
+2. divide systems by maintenance schedules and display a combo box with the names. Let the user choose for which set he want to schedule it. He need to repeat it for the rest.
+3. show the picker multiple times; for every maintenance schedule one.
+
+(1 limiting too much and 3 might look ugly - I would prefer Option 2)
+
+
+### Assign a Maintenance Schedule to a System
+
+There could be multiple ways to assign systems to maintenance schedules.
+
+- In the systems details page we need to show the selected schedule with a view and a change option.
+- It should be possible to change the maintenance schedule for multiple systems using SSM
+- In the details of the Maintenance Schedule we could have a systems page where add/remove is possible.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+
+# Alternatives
+[alternatives]: #alternatives
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+
```

Comment:
What to do with **recurring actions**?
I see two possibilities:
1) Don't allow scheduling recurring events / disable existing events if there is a maintenance window defined.
2) Only execute the (recurring) action if the moment of execution is inside a defined maintenance window.

Reply:
thanks. added